### PR TITLE
Os descriptors upstreaming v4

### PR DESCRIPTION
**Pull request by kopasiak**
Updated version of #15 for community review

### CODE REVIEW

**Comment by falstaff84** [comment]
Tested this patchset with my application which uses RNDIS/OS descriptors, it seems to work fine. Will have a look at the code tomorrow.

File: include/usbg/usbg.h [hunk 1]
```diff
@@ -895,6 +1004,21 @@ extern const char *usbg_get_binding_name(usbg_binding *b);
  */
 extern int usbg_get_binding_name_s(usbg_binding *b, char *buf, int len);
 
+/**
+ * @brief Get configuration selected for OS Descriptors
+ * @param g Pointer to gadget
+ * @return usbg_config or NULL
+ */
+usbg_config *usbg_get_os_desc_binding(usbg_gadget *g);
+
```
Comment (falstaff84) [inline include/usbg/usbg.h]: Nit: There is a superfluous tab here.

File: src/usbg.c [hunk 2]
```diff
@@ -2197,6 +2447,94 @@ int usbg_get_binding_name_s(usbg_binding *b, char *buf, int len)
...
+
+	n = scandir(bpath, &dent, bindings_select, alphasort);
+	if (n < 0) {
+		ret = usbg_translate_error(errno);
+		goto out;
+	}
+
+	if (n < 1) {
+		ret = USBG_ERROR_NOT_FOUND;
+		goto out;
```
Comment (falstaff84) [inline src/usbg.c]: Shouldn't we also free(dent) in this case? The example in https://linux.die.net/man/3/scandir seems to do it.

File: include/usbg/usbg.h (hunk 1 above)
Comment (kopasiak) [reply include/usbg/usbg.h]: Done. Thanks

File: src/usbg.c (hunk 2 above)
Comment (kopasiak) [reply src/usbg.c]: Done. Thanks

**Comment by kopasiak** [comment]
Hi, I fixed your comments and pushed changes once again. Let me know if you find any other issues:)

**Comment by falstaff84** [comment]
Looks good to me!

**Comment by kopasiak** [comment]
Merged. Thank you for your commits, review and especially for your patience:)

**Pull request merged**